src/Core/Silk.NET.Core/Native/SilkMarshal.cs

```diff
@@ -311,7 +328,28 @@ static unsafe string BStrToString(nint ptr)
                 => new string((char*) ptr, 0, (int) (*((uint*) ptr - 1) / sizeof(char)));
 
             static unsafe string AnsiToString(nint ptr) => new string((sbyte*) ptr);
-            static unsafe string WideToString(nint ptr) => new string((char*) ptr);
+
+            static unsafe string WideToString(nint ptr)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    return new string((char*) ptr);
+                }
+                else
+                {
+                    var length = StringLength(ptr, NativeStringEncoding.LPWStr);
```

Comment:
I'm trying to figure out whether doing this is correct.
I've seen references online to using UTF32 when handling wchar_t, such as [Wikipedia](https://en.wikipedia.org/wiki/UTF-32) and [Mono](https://github.com/mono/mono/issues/9362).

Reply:
My main worry is that the old code was casting the `uint` to `ushort`, and considering `wchar_t` is 32-bit for a reason we are likely bugging out on inexpressible characters. I think it's fine to delegate responsibility for wide string decoding to the BCL.

Reply:
Gotcha. My reasoning for casting was because I don't think there will be any data in the 16 bits that are dropped from performing the cast.

I also also built and tested the new package in my project and it works as well so I don't have any complaints here.